benchmark_runner/common/oc/oc.py

```diff
@@ -667,90 +592,230 @@ def wait_for_pod_completed(self, label: str, workload: str = '', label_uuid: boo
-                           namespace: str = environment_variables.environment_variables_dict['namespace'],
-                           timeout: int = int(environment_variables.environment_variables_dict['timeout'])):
+    def delete_vm_sync(self, yaml: str, vm_name: str,
+                       namespace: str = environment_variables.environment_variables_dict['namespace'],
+                       timeout: int = int(environment_variables.environment_variables_dict['timeout'])):
         """
-        Wait for a pod to be ready and running
-        :param pod_name:
+        This method delete vm yaml in async, only if exist and return false if not exist
         :param namespace:
         :param timeout:
-        :return: True if getting pod name or raise PodNameError
+        :param vm_name:
+        :param yaml:
+        :return:
+        """
+        if self.vm_exists(vm_name=vm_name, namespace=namespace):
+            self._delete_async(yaml)
+            return self.wait_for_vm_terminate(vm_name=vm_name, namespace=namespace, timeout=timeout)
+        else:
+            return False
+
+    @logger_time_stamp
+    def wait_for_vm_completed(self, workload: str = '', vm_name: str = '',
+                              namespace: str = environment_variables.environment_variables_dict['namespace'],
+                              timeout: int = int(environment_variables.environment_variables_dict['timeout'])):
+        """
+        This method wait to pod to be completed
+        :param vm_name: The vm name
+        :param workload:
+        :param namespace:
+        :return:
         """
-        self.wait_for_pod_create(pod_name=pod_name, namespace=namespace, timeout=timeout)
         current_wait_time = 0
-        if namespace != '':
-            namespace=f'-n {namespace}'
         while current_wait_time <= timeout:
-            answer = self.run(f'{self.__cli} get pod {namespace} {pod_name} --no-headers -ocustom-columns=Status:status.phase 2>/dev/null')
-            if answer == 'Running':
-                return
-            elif answer == 'Error':
-                raise PodFailed(pod_name)
+            if self.run(
+                    f"{self.__cli} --namespace {namespace} get benchmark {workload} -o jsonpath={{.status.complete}}") == 'true':
+                return True
+            # sleep for x seconds
             time.sleep(OC.SLEEP_TIME)
             current_wait_time += OC.SLEEP_TIME
-        raise PodNotReadyTimeout(pod_name)
+        raise VMNotCompletedTimeout(workload=workload)
+
+    @typechecked
+    @logger_time_stamp
+    def wait_for_vm_terminate(self, vm_name: str,
+                              namespace: str = environment_variables.environment_variables_dict['namespace'],
+                              timeout: int = int(environment_variables.environment_variables_dict['timeout'])):
+        """
+        This method is wait till vm name is terminating or throw exception after timeout
```

Comment:
"Wait for specified VM to terminate; raise VMTerminateTimeout if it does not terminate within the specified timeout".